test/metabase/query_processor_test/expressions_test.clj

```diff
@@ -340,6 +339,77 @@
+;;; - Site locale is irrelevant here.
+
+;;; In these tests, we have a series of specific dates from the checkins table that happen to run Monday to Sunday.
+;;; So in the query results:
+;;; - The **order** of the days is fixed, since we're sorting by date ascending.
+;;; - The **day numbers** are based on the `start-of-week` setting.
+;;; - The **day names** are based on the user locale.
+
+(def ^:private weekdays-english
+  ["Monday" "Tuesday" "Wednesday" "Thursday" "Friday" "Saturday" "Sunday"])
+
+(def ^:private weekdays-spanish
+  ["lunes" "martes" "miércoles" "jueves" "viernes" "sábado" "domingo"])
+
+(deftest ^:synchronized weekday-numbers-and-names-test
+  (mt/test-drivers (mt/normal-drivers-with-feature :expressions)
+    (doseq [first-day                [:sunday :monday :saturday]
+            ;; Adjusting the site locale to get different languages and first day of the week. It should be ignored!
+            site-locale              ["en_US" "es_ES"]
+            ;; For each of two languages and two regions, we check that:
+            ;; 1. the locale's first day of the week is ignored, and
+            ;; 2. the names are correctly translated for the *user* locale.
+            [user-locale exp-names] [;; US uses Sunday=1 in both English and Spanish.
+                                     ["en_US" weekdays-english]
+                                     ["es_US" weekdays-spanish]
+                                                    ;; Europe uses Monday=1 in both English and Spanish.
+                                     ["en_UK" weekdays-english]
+                                     ["es_ES" weekdays-spanish]]]
+      ;; Metabase queries should return weekday numbers based on the *setting*, not the locales.
+      ;; This fixed set of dates from checkins runs Monday to Sunday when sorted ascending by date.
+      (let [known-week  ["2013-02-18"  ; Monday
+                         "2013-02-19"  ; Tuesday
+                         "2013-02-20"  ; Wednesday
+                         "2013-02-21"  ; Thursday
+                         "2013-02-22"  ; Friday
+                         "2013-03-16"  ; Saturday
+                         "2013-03-16"  ; Saturday
+                         "2013-04-28"] ; Sunday
+            ;; The absolute weekdays are fixed above, but the numbering depends on the `start-of-week` setting.
+            exp-numbers (case first-day
+                          :saturday [3 4 5 6 7 1 2]
+                          :sunday   [2 3 4 5 6 7 1]
+                          :monday   [1 2 3 4 5 6 7])]
+        (mt/with-temporary-setting-values [start-of-week first-day
+                                           site-locale   site-locale]
+          (mt/with-user-locale user-locale
+            ;; Fetching [number name date].
+            (let [results (mt/rows (mt/run-mbql-query checkins
+                                     {:fields      [[:expression "weekday"] [:expression "name"] $date]
```

Comment:
Is `$name` included to help debugging? If so, how does it help? If not, why is it needed?

Reply:
It needs to be included because it's in the order-by.

Reply:
What requires this? Can we change it?

Reply:
(Assuming you meant `$date` rather than `$name`.)

Actually it's not required anymore. I was following another example that did this but it's not really needed. I've removed it.